### ISSUE TITLE
Python `rangeQuery`

### ISSUE THREAD

**Issue by bkj** [nmslib/nmslib#117]
Is there a particular reason that `rangeQuery` has not been exposed through the Python bindings?  Or is it just a matter of writing the wrapper?

Thanks
Ben

**Comment by searchivarius** [comment]
It is two things
1) There is no wrapper written (this is an easy part).
2) We don't really have an implementation of the range query in the case of our most efficient methods, i.e., SW-graph, HNSW, and NAPP. So, I didn't find it very useful to create a wrapper. We have been even thinking about removing range queries at all.
Do you find range queries useful?

**Comment by bkj** [comment]
I haven't used the NMSLIB implementation of range queries, but generically I do find them useful -- there are IMO lots of settings where you'd want to find all sufficiently similar datapoints, rather than the top K.

I imagine that though you don't have range query implementations for your most efficient methods, does the NMSLIB implementation still provide significant speedup over brute force for high dimensions, eg between 100 and 5000?

**Comment by searchivarius** [comment]
He Ben, there are ways of implementing the range search. So, I will add this to our list.

Regarding the speed up: it depends on the data (as for nearly any other method). If generate vectors by randomly sampling vector elements from, e.g., U[0,1], I believe that any methods stop working efficiently with about 10-20 dimensions (unless this is a near-duplicate problem).

However, the real data sets are not random. For example, we know that SW-graph, HNSW, and NAPP (with properly generated pivots) work really well for high-dimensional (say 100,000d) *sparse* data.

So, in one of our papes, we were able to efficiently *and accurately* search 1000d vectors even using the VP-tree. 

So, long story short, for many real data sets, there would be substantial speedup, but one needs to run experiments to be sure.

**Comment by gauravahuja** [comment]
Hi, is there an update on this issue?

**Comment by searchivarius** [comment]
@gauravahuja no
